uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -411,9 +412,15 @@ private static function validateLeaveDatesDoesNotOverlapContracts($params) {
       'end_date' => $toDate->format('Y-m-d'),
     ]);
 
-    if ($contract['count'] > 1) {
+    $contractForEndDate = civicrm_api3('HRJobContract', 'getcontractswithdetailsinperiod', [
+      'contact_id' => $params['contact_id'],
+      'start_date' => $toDate->format('Y-m-d'),
+      'end_date' => $toDate->format('Y-m-d'),
+    ]);
+
+    if ($contract['count'] > 1 || $contractForEndDate['count'] != 1) {
       throw new InvalidLeaveRequestException(
-        'The Leave request dates must not have dates in more than one contract period',
+        'This leave request is after your contract end date. Please modify dates of this request',
```

Comment:
Alright, don't you think it would be confusing for the user if they see

> This leave request is after your contract end date.

But the real problem is that the leave request spans two contracts?